### ISSUE TITLE
Forbidden - no checksums provided for signatures

### ISSUE THREAD

**Issue by arichiardi** [slipset/deps-deploy#11]
Thanks @slipset for contacting me for live support on Slack! 

```
Caused by: org.eclipse.aether.transfer.MetadataTransferException: Could not transfer metadata unbroken-promises:unbroken-promises/maven-metadata.xml from/to clojars (https://clojars.org/repo): Access denied to: https://clojars.org/repo/unbroken-promises/unbroken-promises/maven-metadata.xml, ReasonPhrase: Forbidden - no checksums provided for unbroken-promises-0.1.9.xml.asc.
    at org.eclipse.aether.connector.basic.MetadataTransportListener.transferFailed(MetadataTransportListener.java:52)
    at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:364)
    at org.eclipse.aether.connector.basic.BasicRepositoryConnector.put(BasicRepositoryConnector.java:297)
    at org.eclipse.aether.internal.impl.DefaultDeployer.deploy(DefaultDeployer.java:320)
    ... 30 more
Caused by: org.apache.maven.wagon.authorization.AuthorizationException: Access denied to: https://clojars.org/repo/unbroken-promises/unbroken-promises/maven-metadata.xml, ReasonPhrase: Forbidden - no checksums provided for unbroken-promises-0.1.9.xml.asc.
    at org.apache.maven.wagon.shared.http.AbstractHttpClientWagon.put(AbstractHttpClientWagon.java:625)
    at org.apache.maven.wagon.shared.http.AbstractHttpClientWagon.put(AbstractHttpClientWagon.java:557)
    at org.apache.maven.wagon.shared.http.AbstractHttpClientWagon.put(AbstractHttpClientWagon.java:539)
    at org.apache.maven.wagon.shared.http.AbstractHttpClientWagon.put(AbstractHttpClientWagon.java:533)
    at org.apache.maven.wagon.shared.http.AbstractHttpClientWagon.put(AbstractHttpClientWagon.java:513)
    at org.eclipse.aether.transport.wagon.WagonTransporter$PutTaskRunner.run(WagonTransporter.java:653)
    at org.eclipse.aether.transport.wagon.WagonTransporter.execute(WagonTransporter.java:436)
    at org.eclipse.aether.transport.wagon.WagonTransporter.put(WagonTransporter.java:419)
    at org.eclipse.aether.connector.basic.BasicRepositoryConnector$PutTaskRunner.runTask(BasicRepositoryConnector.java:519)
    at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:359)
    ... 32 more
```

This is what I receive now that all the gpg things + Clojars coordinates are in place.

**Comment by arichiardi** [comment]
Maybe @danielcompton can also give us some more insights?

**Comment by slipset** [comment]
With what's on the referenced branch I get:
```
22:17 $  env CLOJARS_USERNAME=erik CLOJARS_PASSWORD=scho5bra clj -A:deploy
gpg passphrase:
Deploying deps-deploy/deps-deploy-0.0.10 to clojars as erik
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Exception in thread "main" org.eclipse.aether.deployment.DeploymentException: Failed to deploy metadata: Could not transfer metadata deps-deploy:deps-deploy/maven-metadata.xml from/to clojars (https://clojars.org/repo): Access denied to: https://clojars.org/repo/deps-deploy/deps-deploy/maven-metadata.xml, ReasonPhrase: Forbidden - deps-deploy-0.0.10.pom has no signature.
	at org.eclipse.aether.internal.impl.DefaultDeployer.deploy(DefaultDeployer.java:326)
	at org.eclipse.aether.internal.impl.DefaultDeployer.deploy(DefaultDeployer.java:223)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.deploy(DefaultRepositorySystem.java:384)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at clojure.lang.Reflector.invokeMatchingMethod(Reflector.java:167)
	at clojure.lang.Reflector.invokeInstanceMethod(Reflector.java:102)
	at cemerick.pomegranate.aether$deploy_artifacts.invokeStatic(aether.clj:358)
	at cemerick.pomegranate.aether$deploy_artifacts.doInvoke(aether.clj:308)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at clojure.core$apply.invoke(core.clj:660)
	at cemerick.pomegranate.aether$deploy.invokeStatic(aether.clj:427)
	at cemerick.pomegranate.aether$deploy.doInvoke(aether.clj:391)
	at clojure.lang.RestFn.invoke(RestFn.java:512)
	at deps_deploy.deps_deploy$eval1650$fn__1652.invoke(deps_deploy.clj:76)
	at clojure.lang.MultiFn.invoke(MultiFn.java:229)
	at deps_deploy.deps_deploy$_main.invokeStatic(deps_deploy.clj:89)
	at deps_deploy.deps_deploy$_main.doInvoke(deps_deploy.clj:87)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.lang.Var.applyTo(Var.java:705)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at clojure.main$main_opt.invokeStatic(main.clj:491)
	at clojure.main$main_opt.invoke(main.clj:487)
	at clojure.main$main.invokeStatic(main.clj:598)
	at clojure.main$main.doInvoke(main.clj:561)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.lang.Var.applyTo(Var.java:705)
	at clojure.main.main(main.java:37)
Caused by: org.eclipse.aether.transfer.MetadataTransferException: Could not transfer metadata deps-deploy:deps-deploy/maven-metadata.xml from/to clojars (https://clojars.org/repo): Access denied to: https://clojars.org/repo/deps-deploy/deps-deploy/maven-metadata.xml, ReasonPhrase: Forbidden - deps-deploy-0.0.10.pom has no signature.
	at org.eclipse.aether.connector.basic.MetadataTransportListener.transferFailed(MetadataTransportListener.java:52)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:364)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.put(BasicRepositoryConnector.java:297)
	at org.eclipse.aether.internal.impl.DefaultDeployer.deploy(DefaultDeployer.java:320)
	... 30 more
Caused by: org.apache.maven.wagon.authorization.AuthorizationException: Access denied to: https://clojars.org/repo/deps-deploy/deps-deploy/maven-metadata.xml, ReasonPhrase: Forbidden - deps-deploy-0.0.10.pom has no signature.
	at org.apache.maven.wagon.shared.http.AbstractHttpClientWagon.put(AbstractHttpClientWagon.java:625)
	at org.apache.maven.wagon.shared.http.AbstractHttpClientWagon.put(AbstractHttpClientWagon.java:557)
	at org.apache.maven.wagon.shared.http.AbstractHttpClientWagon.put(AbstractHttpClientWagon.java:539)
	at org.apache.maven.wagon.shared.http.AbstractHttpClientWagon.put(AbstractHttpClientWagon.java:533)
	at org.apache.maven.wagon.shared.http.AbstractHttpClientWagon.put(AbstractHttpClientWagon.java:513)
	at org.eclipse.aether.transport.wagon.WagonTransporter$PutTaskRunner.run(WagonTransporter.java:653)
	at org.eclipse.aether.transport.wagon.WagonTransporter.execute(WagonTransporter.java:436)
	at org.eclipse.aether.transport.wagon.WagonTransporter.put(WagonTransporter.java:419)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$PutTaskRunner.runTask(BasicRepositoryConnector.java:519)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:359)
	... 32 more
```
It's approaching bed time here, so I'll have to dig deeper tomorrow

**Comment by danielcompton** [comment]
@arichiardi what would you like help with specifically? The error messages seem to suggest the problem is missing signatures/checksums? Are the checksums and signatures being sent? Is it possible to log the files as they are being deployed? I can't see that in either of the logs included here. Leiningen prints that out while it is deploying which can be helpful when debugging these kinds of issues.

**Comment by arichiardi** [comment]
@danielcompton you are right, we can maybe output that. I was wondering who generates that checksum but I can dig into `pomegranade` :smile: 
Thanks!

**Comment by cch1** [comment]
What's the status of this issue?  I've got the same problem.

**Comment by cch1** [comment]
Here's some relevant background: https://github.com/clojars/clojars-web/issues/598

**Comment by cch1** [comment]
Determined that the bug is only present when signing jars.

**Comment by slipset** [comment]
@cch1 Thanks for nudging me here! I finally figured out the problem. `pom.xml` needs to be renamed to `deps-deploy-0.1.2.pom` before sending.

**Comment by cch1** [comment]
Yep, that’s why I figured out as well by process of elimination.

Chris Hapgood
cch1@hapgood.com

**Comment by slipset** [comment]
@cch1 released `0.1.3` which seems to Do The Right Thing. Enjoy
